File: codewars/8kyu/string-repeat.py
# SOLUTION
# ==============================================================================
def repeat_str(repeat, string):
    return string * repeat

# TESTS
# ==============================================================================
if __name__ == "__main__":
    assert repeat_str(4, "a") == "aaaa", "Test case 1 failed"
    assert repeat_str(3, "abc") == "abcabcabc", "Test case 2 failed"
    assert repeat_str(0, "hello") == "", "Test case 3 failed"
    assert repeat_str(2, "") == "", "Test case 4 failed"
    assert repeat_str(5, "xyz") == "xyzxyzxyzxyzxyz", "Test case 5 failed"
    print("All test cases passed!")

File: codewars/8kyu/string-repeat.cpp
#include <iostream>
#include <string>
#include <cassert>

std::string repeat_str(size_t repeat, const std::string &str) {
    std::string result{};
    for (int i = 0; i < repeat; ++i) {
        result.append(str);
    }
    return result;
}

void runTests() {
    assert(repeat_str(3, "*") == "***");
    assert(repeat_str(5, "#") == "#####");
    assert(repeat_str(2, "ha ") == "ha ha ");
    std::cout << "All tests passed!" << '\n';
}

int main() {
    runTests();
    return 0;
}
